src/plugins/data_views/server/register_index_pattern_usage_collection.ts

```diff
@@ -80,60 +85,74 @@ export async function getIndexPatternTelemetry(indexPatterns: DataViewsContract)
     },
   };
 
-  await ids.reduce(async (col, id) => {
-    await col;
-    const ip = await indexPatterns.get(id);
+  const findOptions: SavedObjectsCreatePointInTimeFinderOptions = {
+    type: DATA_VIEW_SAVED_OBJECT_TYPE,
+    perPage: 1000,
+    fields: ['attributes.fields', 'attributes.runtimeFieldMap'],
+  };
 
-    const scriptedFields = ip.getScriptedFields();
-    const runtimeFields = ip.fields.filter((fld) => !!fld.runtimeField);
+  const finder = savedObjectsService.createPointInTimeFinder<DataViewFieldAttrs>(findOptions);
 
-    if (scriptedFields.length > 0) {
-      // increment counts
-      results.indexPatternsWithScriptedFieldCount++;
-      results.scriptedFieldCount += scriptedFields.length;
+  for await (const response of finder.find()) {
+    const { saved_objects: savedObjects, total } = response;
+    results.indexPatternsCount = total;
 
-      // calc LoC
-      results.perIndexPattern.scriptedFieldLineCount = minMaxAvgLoC(
-        scriptedFields.map((fld) => fld.script || '')
-      );
-
-      // calc field counts
-      results.perIndexPattern.scriptedFieldCount.min = updateMin(
-        results.perIndexPattern.scriptedFieldCount.min,
-        scriptedFields.length
-      );
-      results.perIndexPattern.scriptedFieldCount.max = updateMax(
-        results.perIndexPattern.scriptedFieldCount.max,
-        scriptedFields.length
-      );
-      results.perIndexPattern.scriptedFieldCount.avg =
-        results.scriptedFieldCount / results.indexPatternsWithScriptedFieldCount;
-    }
-
-    if (runtimeFields.length > 0) {
-      // increment counts
-      results.indexPatternsWithRuntimeFieldCount++;
-      results.runtimeFieldCount += runtimeFields.length;
+    savedObjects.forEach((obj) => {
+      const fields = JSON.parse(obj.attributes.fields) || [];
+      const runtimeFieldsMap = obj.attributes.runtimeFieldMap
+        ? JSON.parse(obj.attributes.runtimeFieldMap) || {}
```

Comment:
This is also a candidate for `try - catch`

Reply:
@mattkime Telemetry stats now work as excepted 👍  ... one think I still think worth to consider is using a `try - catch` pattern for the JSON.parse. Because if for whatever reason an invalid JSON string is passed in, a `SyntaxError` is thrown, and this would lead and the one hostile data view would stop the generation of the results ... or do I miss somewhere this was caught? ... or would such a case also lead to troubles before this change?